### DISCUSSION TITLE
Accelerate Development with LLM

### DISCUSSION THREAD

**Discussion by davidjadczyk** [davidmigloz/langchain_dart#17] [Ideas]
LLMs like GPT3.5, and GPT 4 event better, are great at understanding code. Thus, they can easily transform a code section from one language to another. As there are already Gateways like to langchainjs it could be beneficial to rewrite that codebase with the help of LLMs to dart. This could be especially beneficial, when there are very few contributors, to still accelerate the development of this project.

**Comment by davidmigloz** [comment]
Hi @DJ2695 ,

Thanks for your suggestion.

Indeed, I'm using GPT-4 and Copilot to help me speed up the process of porting the library. While I am getting good results for small blocks of code (e.g. rewriting the logic of a function or porting unit tests), I'm not managing to get satisfactory results while trying to convert bigger parts (e.g. a full file with several classes). Python (and TypeScript) are very permissive languages and unfortunately, the original library relies a lot on this (e.g. in the Python package they use `kwargs` to pass arbitrary data all over the place, in the TS package they rely on union types substantially), which makes the translation to Dart harder. Also, while I'm trying to follow the original API as closely as possible, I'm also trying to make it as much idiomatic Dart (v3) as possible, which is not something that GPT-4 is very good at (yet) without spending too much effort on prompting.

But if you have any suggestions on how to improve the process, I'm all ears!
